pkg/persistence/persistence_impl.go

```diff
@@ -0,0 +1,66 @@
+package persistence
+
+import "fmt"
+
+// PersistableID is the type of the UUID of a persistable object.
+type PersistableID string
```

Comment:
Nitpick: The package name defines a scope and therefore I would recommend to remove the prefix `Persistable` in all the Type names. Eventually it'll be `persistence.ID` ...